tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -308,7 +308,8 @@ GpuExecutable::ResolveConstantGlobals(se::Stream* stream) {
 }
 
 StatusOr<se::DeviceMemoryBase> GpuExecutable::BufferForAllocation(
-    absl::Span<ExecutionInput const> arguments,
+    absl::Span<const ShapedBuffer* const>* arguments_buffer,
+    std::vector<ExecutionInput>* arguments_inputs,
```

Comment:
Here I used pointers as this is what I know more. But I could have used 2 std::optional or 1 std::variant.
If you want me to change the code to use one of them, tell me.

Reply:
As MaybeOwningDeviceMemory is just a std::variant as you said, then I suppose I can't use std::variant here. It won't keep the speed up we are looking for. Do you think using std::optional would be faster then std::variant? I do not know their implementation. If you do not know, for the speed, I'll investigate it.

Reply:
Forget the comment above. MaybeOwningDeviceMemory is an std::variant per argument.
Doing an std::variant for the span/vector shouldn't be cheaper then doing it per elements.
So if you agree with this PR, I'll update it to use and std::variant here instead of 2 ptr.